### DISCUSSION TITLE
A raw string, or heredoc style syntax

### DISCUSSION THREAD

**Discussion by itsgreggreg** [mint-lang/mint#379] [Language Development]
Would be nice to have a raw string syntax that consumed everything between its delimiters literally including quotes and newlines. Helps with including strings that would otherwise need a lot of escapes such as JSON strings or human writing. 

Would be nice if it is the case that directly after the opening delimiter there is a newline, that newline isn't included in the output.  Similarly if there is no content between the last newline and the closing delimiter, then that last newline is not included in the output.

So this string would contain exactly 1 new line character:
```
"""
one
two
"""
```

Could also be nice to have some sort of white space removal indicator, perhaps the location of the closing delimiter like in Crystal. So the const here would have no leading whitespace at runtime:
```
module A {
    const a = """
        { "key": "val" }
        """
}
```

Syntax examples from other languages:

### Elm
```elm
"""
{
  "json": "without escapes
}
"""
```

### Elixir
```elixir
~S"""
{
  "json": "without escapes
}
"""
```

### Rust
```rust
r"
{
  "json": "without escapes
}
"
```

### Crystal
```crystal
<<-JSON
{
  "json": "without escapes
}
JSON
```

**Comment by gdotdesign** [comment]
I think it would be a good addition and it's not hard to implement personally I vote for the the triple quote syntax `"""` .

**Comment by Sija** [comment]
HEREDOC strings were implemented in newly released version 0.17.0.
